### DISCUSSION TITLE
Dropping Livewire

### DISCUSSION THREAD

**Discussion by Propaganistas** [opcodesio/log-viewer#21] [Ideas]
This Log Viewer looks promising. Yet I'm reluctant to install it because it'll pull in Livewire as an additional dependency.
Any chance to drop Livewire and keep this package sort of self-contained similar to Laravel Horizon, Telescope, ...?

**Comment by arukompas** [comment]
Not in the short-term future, unfortunately. It would be a significant rework to change it to Vue or some other SPA framework without pulling any additional PHP libs.

Is Livewire conflicting with something for you? What is the reason you don't want that dependency? 🤔

**Comment by Propaganistas** [comment]
I don't suggest using another SPA framework either - just blade files and JS (with jQuery or Alpine as DOM enhancer), similar to what Horizon offers.

For me it's a personal preference to keep my project's (sub)dependency list as small as possible. But I can imagine that version conflicts might occur for projects pulling in livewire as well, as you're basically forcing projects to use "your" version of livewire if they want to use your package.

**Comment by arukompas** [comment]
Makes total sense. I'll definitely keep this in mind, but I don't think I'll rewrite it in the near future. Perhaps for v2
